### ISSUE TITLE
Clarify deploy --tag option usage

### ISSUE THREAD

**Issue by erno** [aws/copilot-cli#4973]
Hello,

I was looking into how to deploy a specified tag (git hash for us) of a service, when loading the image from a registry. After some testing (no change in deployed image according to ECS console) I started wondering why the svc deploy doc says it can "Tag it with the value from --tag or the latest git sha (if you're in a git directory)" and now my best guess is that this option actually only has an effect when building a local Docker image. If so, the doc & help string should be updated to say this.

**Comment by iamhopaul123** [comment]
Hello @erno. Yes you are correct. `--tag` is used for tagging the image we help build from local dockerfile. If you have an existing registry and you want to use image within with specific tag, you can just refer it in you [manifest](https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#image-location). However, I do agree with you that the docstring is not accurate enough. Thank you for letting us know!

Edited: Actually after taking another look at our doc:

```
`copilot svc deploy` takes your local code and configuration and deploys it.

The steps involved in service deploy are:

1. Build your local Dockerfile into an image
2. Tag it with the value from `--tag` or the latest git sha (if you're in a git directory)
3. Push the image to ECR
4. Package your manifest file and addons into CloudFormation
4. Create / update your ECS task definition and service
```

It is to explain what Copilot will do under the hood if you run `svc deploy` and in step 1 it specifies Copilot will build local dockerfile into an image. Is it still unclear to you 🤔

**Comment by erno** [comment]
Hi,

What's unclear IMO is copilot svc deploy --help output which says only:
"--tag string                     Optional. The container image tag"

The steps list seems something to skip over when looking for information about "what does --tag do", it's a usage example and not a reference style doc piece.

Also isn't the steps doc also out of date because it says in step 1 that the image is always locally built?

**Comment by iamhopaul123** [comment]
Hello @erno. Yes, that's a good callout. Thank you for pointing that out! I'll update the doc accordingly. In the meantime, please let us know if anything is still not clear to you.

**Comment by huanjani** [comment]
The documentation update has been released in v1.29.0: https://github.com/aws/copilot-cli/releases/tag/v1.29.0! Thanks for the feedback.
